spylib/admin_api.py

```diff
@@ -196,16 +199,22 @@ async def execute_gql(
 
         # Handle any response that is not 200, which will return with error message
         # https://shopify.dev/api/admin-graphql#status_and_error_codes
+        if resp.status_code in [500, 503]:
+            raise ShopifyIntermittentError(
+                f'The Shopify API returned an intermittent error: {resp.status_code}.'
+            )
+
         if resp.status_code != 200:
-            jsondata = resp.json()
-            if jsondata:
-                error_msg = jsondata['errors']
-                raise ShopifyGQLError(
-                    f'GQL query failed, status code: {resp.status_code}. {error_msg}'
-                )
-            else:
+            try:
+                jsondata = resp.json()
+            except JSONDecodeError:
                 raise ShopifyGQLError(f'GQL query failed, status code: {resp.status_code}.')
 
+            error_msg = jsondata['errors']
+            raise ShopifyGQLError(
+                f'GQL query failed, status code: {resp.status_code}. {error_msg}'
+            )
```

Comment:
issue: it feels like it can be simpler and more readable by highlighting the difference between having and not having the json data:
```suggestion
            try:
                jsondata = resp.json()
                error_msg = f'{resp.status_code}. {jsondata["errors"]}'
            except JSONDecodeError:
                error_msg = f'{resp.status_code}.'

            raise ShopifyGQLError(f'GQL query failed, status code: {error_msg}')
```